server/src/main/java/org/apache/iotdb/db/service/IoTDB.java

```diff
@@ -123,7 +120,22 @@ private void setUp() throws StartupException {
     if (IoTDBDescriptor.getInstance().getConfig().isEnableMQTTService()) {
       registerManager.register(MQTTService.getInstance());
     }
-    logger.info("IoTDB is set up.");
+
+    logger.info("IoTDB is set up, now may some sgs are not ready, don't worry, the superman Dr. Huang will teach the sg to fix themselves.");
+
+    while (!StorageEngine.getInstance().isAllSgReady()) {
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
+      }
+    }
```

Comment:
How about just waiting where you created that single thread pool and shuting down it?

Reply:
Good idea!